### DISCUSSION TITLE
The required column `cuid` was added to the `MyModel` table with a prisma-level default value. There are 13 rows in this table, it is not possible to execute this step. Please add this column as optional, then populate it before making it required.

### DISCUSSION THREAD

**Discussion by HartS** [prisma/prisma#11035] [Q&A] [answered]
I've definitely added default values on tables in the past, why is it failing now? Is it related to the number of records? Searching for this error doesn't give me any google results

**Comment by janpio** [comment]
What is giving you this error message?
What version of Prisma are you using?

**Reply by HartS** [reply]
This is prisma 3.6.0; I've added a new field to a model with `@default(cuid())`. Unfortunately I'm not able to apply this to the database; `prisma db push` gives me this error. I tried with the field being both required and optional. It seems like the workaround will be to make the field an optional field with no default, generate and update the cuids with a script, and then hopefully I can add the default after.

Whatever is causing this doesn't seem to be documented anywhere in the prisma docs. Not sure if it matters, but I'm using sqlite also.

**Reply by janpio** [reply]
> It seems like the workaround will be to make the field an optional field with no default, generate and update the cuids with a script, and then hopefully I can add the default after.

Yes, isn't this what the error message is telling you to do with `Please add this column as optional, then populate it before making it required`?

**Reply by HartS** [reply]
Sure, that 'works', but I don't get prisma's cuid generation (I need to add a package and script to insert cuids myself). My question is about why this doesn't work with a default value... I'm fairly certain I've done this before, with cuid() even.

**Reply by janpio** [reply, marked as answer]
The only way this works is if there is no existing data in the table. Prisma has not concept to backfill data like this.

**Reply by ale-diez** [reply]
Do you have a suggestion on how to address this use case, since at the moment Prisma only handles raw SQL migrations and PostgreSQL doesn't support cuids natively?

**Reply by HartS** [reply]
@demedos I settled on building a migration system around Prisma that would allow running bash scripts (which themselves can alternate between running prisma migrations, and JS scripts).

In this case I added a script to migrate the schema to add an optional cuid field, run a JS script which populated the records in the table using the cuid library (make sure you're using the version of cuid supported by your version of prisma), and call prisma migration again to mark the field required

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
